roles/contiv/tasks/download_bins.yml

```diff
@@ -25,3 +25,20 @@
     src: "{{ contiv_current_release_directory }}/netplugin-{{ contiv_version }}.tar.bz2"
     dest: "{{ contiv_current_release_directory }}"
     copy: no
+
+- name: Download Bins | Download cni tar file
+  get_url:
+    url: "{{ cni_bin_url }}"
+    dest: "{{ cni_download_dir }}"
+    mode: 0755
+    validate_certs: False
+  environment:
+    http_proxy: "{{ http_proxy|default('') }}"
+    https_proxy: "{{ https_proxy|default('') }}"
+    no_proxy: "{{ no_proxy|default('') }}"
+
+- name: Download Bins | Extract cni tar file
+  unarchive:
+    src: "{{ cni_download_dir }}/cni-{{ cni_version }}.tbz2"
```

Comment:
@srampal the `src` of the `unarchive` action could be a variable that was registered from `get_url`, e.g.

```
- name: Download Bins | Download cni tar file
  get_url:
    url: "{{ cni_bin_url }}"
    dest: "{{ cni_download_dir }}"
    mode: 0755
    validate_certs: False
  environment:
    http_proxy: "{{ http_proxy|default('') }}"
    https_proxy: "{{ https_proxy|default('') }}"
    no_proxy: "{{ no_proxy|default('') }}"
  register: download_file

- name: Download Bins | Extract cni tar file
  unarchive:
    src: "{{ download_file.dest }}"
    dest: "{{ cni_download_dir }}"
    copy: no
  when: download_file.changed
```
Also adding a when clause is a further step to idempotent playbook.